src/RBACHandler.ts

```diff
@@ -20,36 +33,99 @@ export class RBACHandler implements Authorization {
+    isAccessBulkDataJobAllowed(request: AccessBulkDataJobRequest): void {
+        if (request.requesterUserId !== request.jobOwnerId) {
+            throw new UnauthorizedError('Unauthorized');
+        }
     }
 
-    async isBundleRequestAuthorized(request: AuthorizationBundleRequest): Promise<boolean> {
-        const decoded = decode(request.accessToken, { json: true }) || {};
+    private isBulkDataAccessAllowed(groups: string[], bulkDataAuth: BulkDataAuth): void {
+        const { operation, exportType } = bulkDataAuth;
+        if (['get-status-export', 'cancel-export', 'get-status-import', 'cancel-import'].includes(operation)) {
+            return;
+        }
+        if (operation === 'initiate-export') {
+            for (let index = 0; index < groups.length; index += 1) {
+                const group: string = groups[index];
+                if (this.rules.groupRules[group]) {
+                    const rule: Rule = this.rules.groupRules[group];
+                    if (exportType && rule.operations.includes('read')) {
+                        if (exportType === 'system') {
+                            // TODO: Enable supporting of different profiles by specifying the resources you would want to export
+                            // in BASE_R4_RESOURCES
+                            if (
+                                (this.fhirVersion === '4.0.1' &&
+                                    isEqual(rule.resources.sort(), BASE_R4_RESOURCES.sort())) ||
+                                (this.fhirVersion === '3.0.1' &&
+                                    isEqual(rule.resources.sort(), BASE_STU3_RESOURCES.sort()))
+                            ) {
+                                return;
+                            }
+                        }
+                        if (exportType === 'group' || exportType === 'patient') {
+                            let matchEveryResource = false;
+                            if (this.fhirVersion === '4.0.1') {
+                                matchEveryResource = R4_PATIENT_COMPARTMENT_RESOURCES.every((resource: string) => {
+                                    return rule.resources.includes(resource);
+                                });
+                            } else if (this.fhirVersion === '3.0.1') {
+                                matchEveryResource = STU3_PATIENT_COMPARTMENT_RESOURCES.every((resource: string) => {
+                                    return rule.resources.includes(resource);
+                                });
+                            }
+                            if (matchEveryResource) {
+                                return;
+                            }
+                            throw new UnauthorizedError('Unauthorized');
+                        }
+                    }
+                }
+            }
+        } else if (operation === 'initiate-import') {
+            // TODO Handle `initiate-import` auth

```

Comment:
Should we throw a different error in this case?

Reply:
or just 'return' and let the router take care of it?

Reply:
There's currently no set API defined by FHIR for importing data into a server. Once we decide to support import, we would define the API. So as of right now, when we parse the request, we'll never return an initiate-import operation. I don't think that code path will be hit until we start implementing import.